### PR TITLE
Deprecate processorhelper.New*Processor in favor of New*ProcessorWithCreateSettings

### PR DESCRIPTION
**Pull request by bogdandrutu**
Main motivation is to ensure that Settings and Context are passed (to enable obsreport usage which requires settings to be passed, and to enable future logic of cancelation or other things passed in the context).

The plan is next release to deprecate the new func, and change the current func to match the new signature, then in another release to remove the new func.

Signed-off-by: Bogdan <bogdandrutu@gmail.com>

### CODE REVIEW

File: processor/processorhelper/traces.go [hunk 1]
```diff
@@ -36,14 +36,26 @@ type tracesProcessor struct {
...
-// NewTracesProcessor creates a TracesProcessor that ensure context propagation and the right tags are set.
-// TODO: Add observability metrics support
+// Deprecated: [v0.58.0] use version with NewTracesProcessorWithCreateSettings.
 func NewTracesProcessor(
 	cfg config.Processor,
 	nextConsumer consumer.Traces,
 	tracesFunc ProcessTracesFunc,
 	options ...Option,
 ) (component.TracesProcessor, error) {
+	return NewTracesProcessorWithCreateSettings(context.Background(), component.ProcessorCreateSettings{}, cfg, nextConsumer, tracesFunc, options...)
```
Comment (dmitryax) [inline processor/processorhelper/traces.go:46]: Are you planning to bring back `NewTracesProcessor` name once the old one is removed?

Comment (bogdandrutu) [reply processor/processorhelper/traces.go:46]: yes. we need intermediate to keep backwards compatibility for 2 versions.

**Review by dmitryax — APPROVED** [review]
LGTM. But I would like ask for another review from @open-telemetry/collector-approvers since the change is significant

**Comment by bogdandrutu** [comment]
@dmitryax I don't feel is that significant, since we are making the "helper" funcs to look like the Factory funcs (which are the significant once. But more eyes are always welcome.

**Comment by bogdandrutu** [comment]
Per the approvers from https://github.com/open-telemetry/opentelemetry-collector/pull/5834 I take that this is fine as well

**Comment by jpkrohling** [comment]
I haven't reviewed this code in detail, but the general approach is fine.

**Pull request merged**